### PR TITLE
Remove dependencies in test mode for smart widgets

### PR DESCRIPTION
**Pull request by nikolay-vasilev-prime**
#### Overview

> Closes: 700

In order to be able to generate golden tests, we need to exclude the injected dependencies from the views. We created a constant that checks if we are executing from test environment in order to achieve that.

#### Checklist

*Implementation*
- [ ] Implementation matches ticket acceptance criteria and technical notes
- [ ] Manually tested against Acceptance Criteria
- [ ] UI checked in Light / Dark mode

*Stability*
- [ ] Checked if changes affect any features and verified affected features work as expected
- [ ] Added unit tests for new code and verified existing tests work as expected

*Code quality*
- [ ] Updated `CHANGELOG.md`, `README.md` and package versions in `pubspec.yaml`
- [ ] Dependencies are updated to latest versions or new tickets are created if there are breaking changes or deprecations
- [ ] If an unrelated part of the codebase needs to be updated or refactored, create tickets with proposed changes

### CODE REVIEW

File: packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/base/app/config/app_constants.dart [hunk 1]
```diff
@@ -4,8 +4,41 @@
 
 // Here lie all your app constants
 
-/// Vapid Key is required in order to for FCM to run on web properly
-/// https://github.com/FirebaseExtended/flutterfire/blob/4c9b5d28de9eeb5ce76c856fbd0c7b3ec8615e45/docs/messaging/usage.mdx#web-tokens
+/// The [isInTestMode] in these files is used to exclude the dependencies
+/// from smart widgets for golden tests.
+///
+/// Since we don't want to add dependency on [universal_io/io.dart],
```
Comment (StanevPrime) [inline packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/base/app/config/app_constants.dart:10]: I believe this documentation deserves its own place in docs folder  (docs/golden_tests.md) with a reference in the main README.md file (Golden tests section) here we can leave only a reference to the MD file

**Review by StanevPrime — CHANGES_REQUESTED** [review]

**Review by StanevPrime — CHANGES_REQUESTED** [review]

File: packages/rx_bloc_cli/doc/golden_tests.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+            providers: [
+                ..._services,
+                ..._blocs,
+            ],
+            child: current,
+        );
+    }
+}
+```
+IMPORTANT: you should import `isInTestMode` from [app_constants.dart]!
```
Comment (DDavidPrime) [inline packages/rx_bloc_cli/doc/golden_tests.md:30]: You can use specific markdown formatting to indicate[ different levels of alerts](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts).

File: packages/rx_bloc_cli/doc/golden_tests.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
+### Smart Widgets dependencies for Golden Tests
+
+In golden tests we want to mock the BLoCs to be able to test all UI states.Using reusable smart widgets that have their own dependency injection will make golden tests to fail. To fix this issue we need to build the smart widgets without injecting the required components during the test. This is why we added `isInTestMode` in `app_constants.dart`.
```
Comment (DDavidPrime) [inline packages/rx_bloc_cli/doc/golden_tests.md:3]: A cosmetical suggestion.

We can rephrase the second sentence to something like this:
`By using smart widgets that manage their dependencies, golden tests could fail.`

**Review by DDavidPrime — APPROVED** [review]

File: packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/base/app/config/app_constants.dart (hunk 1 above)
Comment (nikolay-vasilev-prime) [reply packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/base/app/config/app_constants.dart]: Done

File: packages/rx_bloc_cli/mason_templates/bricks/lib_auth_matrix/__brick__/lib/lib_auth_matrix/di/auth_matrix_otp_page_with_dependencies.dart [hunk 4]
```diff
@@ -4,6 +4,7 @@ import 'package:flutter/cupertino.dart';
 import 'package:flutter_rx_bloc/flutter_rx_bloc.dart';
 import 'package:provider/provider.dart';
 
+import '../../base/app/config/app_constants.dart';
```
[the remark this answers is not in the document]
Comment (nikolay-vasilev-prime) [reply packages/rx_bloc_cli/mason_templates/bricks/lib_auth_matrix/__brick__/lib/lib_auth_matrix/di/auth_matrix_otp_page_with_dependencies.dart]: Done

File: packages/rx_bloc_cli/mason_templates/bricks/lib_auth_matrix/__brick__/lib/lib_auth_matrix/di/auth_matrix_pin_biometrix_page_with_dependencies.dart [hunk 5]
```diff
@@ -4,6 +4,7 @@ import 'package:flutter/cupertino.dart';
 import 'package:flutter_rx_bloc/flutter_rx_bloc.dart';
 import 'package:provider/provider.dart';
 
+import '../../base/app/config/app_constants.dart';
```
[the remark this answers is not in the document]
Comment (nikolay-vasilev-prime) [reply packages/rx_bloc_cli/mason_templates/bricks/lib_auth_matrix/__brick__/lib/lib_auth_matrix/di/auth_matrix_pin_biometrix_page_with_dependencies.dart]: Done

File: packages/rx_bloc_cli/doc/golden_tests.md (hunk 2 above)
Comment (nikolay-vasilev-prime) [reply packages/rx_bloc_cli/doc/golden_tests.md]: Done

File: packages/rx_bloc_cli/doc/golden_tests.md (hunk 3 above)
Comment (nikolay-vasilev-prime) [reply packages/rx_bloc_cli/doc/golden_tests.md]: Done

File: packages/rx_bloc_cli/mason_templates/bricks/lib_auth_matrix/__brick__/lib/lib_auth_matrix/di/auth_matrix_pin_biometrix_page_with_dependencies.dart [hunk 6]
```diff
@@ -10,10 +10,10 @@ import '../views/auth_matrix_pin_biometrics_page.dart';
 
 class AuthMatrixPinBiometricsPageWithDependencies extends StatelessWidget {
   const AuthMatrixPinBiometricsPageWithDependencies({
+    Key? key,
     required this.response,
     required this.endToEndId,
-    super.key,
-  });
+  }) : super(key: key);
```
Comment (StanevPrime) [inline packages/rx_bloc_cli/mason_templates/bricks/lib_auth_matrix/__brick__/lib/lib_auth_matrix/di/auth_matrix_pin_biometrix_page_with_dependencies.dart:16]: Remo

**Review by StanevPrime — CHANGES_REQUESTED** [review]
All auth-matrix related changes need to be removed

File: packages/rx_bloc_cli/doc/golden_tests.md
Comment (StanevPrime) [inline packages/rx_bloc_cli/doc/golden_tests.md:1]: Please make sure this file is copied in the example folder, the same as https://github.com/Prime-Holding/rx_bloc/blob/develop/packages/rx_bloc_cli/example/docs/continuous_delivery.md

File: packages/rx_bloc_cli/README.md [hunk 7]
```diff
@@ -96,7 +96,7 @@ Out of the box, a Rx Bloc CLI created projects includes:
 
 ✅ [Design system][design_system_lnk] - A single place where you can define all your colors, typography, assets and more. Your app's Light and Dark mode are already configured
 
-✅ [Testing][testing_lnk] - With Unit and Golden Tests you know your app is working as intended
+✅ [Testing][testing_lnk] - With Unit and Golden Tests you know your app is working as intended. Please read the Rx Bloc CLI specific [Golden Tests documentation][golden_tests_documentation_lnk].
```
Comment (StanevPrime) [inline packages/rx_bloc_cli/README.md:99]: the change log and the version in pubspec needs to be updated. make sure [rxBlocCliPackageVersion constant](https://github.com/Prime-Holding/rx_bloc/blob/develop/packages/rx_bloc_cli/lib/src/rx_bloc_cli_constants.dart#L2) is updated to reflect the current version

**Review by StanevPrime — CHANGES_REQUESTED** [review]

File: packages/rx_bloc_cli/doc/golden_tests.md
Comment (nikolay-vasilev-prime) [reply packages/rx_bloc_cli/doc/golden_tests.md:1]: Copied or moved?

Comment (DDavidPrime) [reply packages/rx_bloc_cli/doc/golden_tests.md:1]: Usually, before we push changes to the remote repo, we run the [`compile_test_project.sh`](https://github.com/Prime-Holding/rx_bloc/blob/develop/packages/rx_bloc_cli/bin/compile_test_project.sh) script which prepares the content for us from the generated example project. This is a good starting point for what you'll need.

Comment (nikolay-vasilev-prime) [reply packages/rx_bloc_cli/doc/golden_tests.md:1]: Done

File: packages/rx_bloc_cli/README.md (hunk 7 above)
Comment (nikolay-vasilev-prime) [reply packages/rx_bloc_cli/README.md:99]: Done

File: packages/rx_bloc_cli/CHANGELOG.md [hunk 8]
```diff
@@ -1,3 +1,6 @@
+## [3.7.1]
+Added functionality to exclude smart widgets dependency injection components during golden tests.
```
Comment (StanevPrime) [inline packages/rx_bloc_cli/CHANGELOG.md:2]: Please follow the convention of the changelog verions

**Review by StanevPrime — CHANGES_REQUESTED** [review]

**Review by StanevPrime — APPROVED** [review]

**Pull request merged**